src/runner/pythonrunner.cpp

```diff
@@ -1445,11 +1458,56 @@ QList<QObject*> PythonRunner::instantiate(const QString &pluginName)
     return allInterfaceList;
   }
   catch (const bpy::error_already_set&) {
-    MOBase::log::error("Failed to import plugin from {}.", pluginName);
+    MOBase::log::error("Failed to import plugin from {}.", identifier);
     throw pyexcept::PythonError();
   }
 }
 
+void PythonRunner::unload(const QString& identifier)
+{
+  auto it = m_PythonObjects.find(identifier);
+  if (it != m_PythonObjects.end()) {
+
+    GILock lock;
+
+    if (!identifier.endsWith(".py")) {
```

Comment:
This handles the case of a folder, while in the case of a single .py file in top level? Is it handled with the map of objects later?

Reply:
If you have a single `.py`, you don't have a module loaded for this plugin, the file is simply executed, so there is nothing to remove. The plugin will be released when deleting the object and when re-loading, the file will be re-executed.

This does not handle `.py` file with stuff in `data/` since it's impossible to know from which plugins the file in `data/` come, but I'd say that should not be used with newest MO2 version (even if we don't force a folder per extension/plugin).

Reply:
Ok, thanks for explainig!